### PR TITLE
Fix TLS handshake

### PR DESCRIPTION
**Pull request by gregschohn**
### Description
Better reliability for the replayer to make send TLS traffic to target clusters!
* **_Category_** Bug fix
* **_Why these changes are required?_** Previously, some data was sent unencrypted and wasn't properly parsed or handled by the target cluster
* **_What is the old behavior before changes and new behavior after changes?_**
 * Old results - about 87-98% accuracy on geonames and log OpenSearch benchmark datasets.
 * New results - about 100% accuracy on geonames and log OpenSearch benchmark datasets.

### Issues Resolved
https://opensearch.atlassian.net/browse/MIGRATIONS-1190 

### Testing
dockerSolution

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by mikaylathompson** [comment]
Woah! 100% accuracy is huge!

File: TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/NettyPacketToHttpConsumer.java [hunk 1]
```diff
@@ -2,16 +2,17 @@
 
 import io.netty.bootstrap.Bootstrap;
 import io.netty.buffer.ByteBuf;
-import io.netty.channel.Channel;
-import io.netty.channel.ChannelFuture;
-import io.netty.channel.ChannelFutureListener;
-import io.netty.channel.ChannelOption;
+import io.netty.channel.*;
```
Comment (lewijacn) [inline TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/NettyPacketToHttpConsumer.java:5]: nit: Normally against .*

File: TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/NettyPacketToHttpConsumer.java [hunk 2]
```diff
@@ -45,112 +85,120 @@ public NettyPacketToHttpConsumer(NioEventLoopGroup eventLoopGroup, URI serverUri
...
+                    pipeline.addLast(new LoggingHandler("POST_SSL", LogLevel.TRACE));
+                    pipeline.addLast(new SslErrorHandler(e-> terminateChannel(pipeline)));
+                    pipeline.addLast(new LoggingHandler("POST_ERROR_CATCHER", LogLevel.TRACE));
+                    sslHandler.handshakeFuture().addListener(f -> {
+                        if (f.isSuccess()) {
+                            addHttpTransactionHandlersToPipeline(pipeline);
+                            fullyInitializedChannelFuture.future.complete(future.channel());
+                        } else {
+                            fullyInitializedChannelFuture.future.completeExceptionally(future.cause());
+                        }
```
Comment (lewijacn) [inline TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/NettyPacketToHttpConsumer.java:114]: Is this essentially saying that if my ssl handshake fails I will not add further handlers to the pipeline? Are there existing handlers that this message could still go to?

**Review by lewijacn — APPROVED** [review]

File: TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/NettyPacketToHttpConsumer.java (hunk 1 above)
Comment (gregschohn) [reply TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/NettyPacketToHttpConsumer.java:5]: Me too!  Different IDE that I didn't correct yet.

File: TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/NettyPacketToHttpConsumer.java (hunk 2 above)
Comment (gregschohn) [reply TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/NettyPacketToHttpConsumer.java:119]: The only other ones in the pipeline at this point are diagnostic ones.  addHttpTransactionHandlersToPipeline adds the other ones, but only after a successful handshake

**Pull request merged**